saw-core-what4/src/Verifier/SAW/Simulator/What4/ReturnTrip.hs

```diff
@@ -367,6 +367,15 @@ scEq sym sc tp x y =
     BaseBoolRepr    -> scBoolEq sc x y
     BaseRealRepr    -> scRealEq sym sc x y
     BaseIntegerRepr -> scIntEq sc x y
+    BaseArrayRepr idxTypes range
+      | Ctx.Empty Ctx.:> idx_type <- idxTypes ->
+        do let SAWExpr x' = x
+           let SAWExpr y' = y
+           sc_idx_type <- baseSCType sym sc idx_type
+           sc_elm_type <- baseSCType sym sc range
+           SAWExpr <$> SC.scArrayEq sc sc_idx_type sc_elm_type x' y'
+      | otherwise ->
+        unsupported sym ("SAW backend: equality comparison on unsupported multidimensional array type:" ++ show tp)
```

Comment:
Is the reason not to support multi-dimensional arrays here that it would be extra work, in this function, that we don't need right now? Or are there limitations elsewhere in the codebase that would make it hard to support them?